src/chain/chain_accumulator.rs

```diff
@@ -98,6 +150,45 @@ impl ChainAccumulator {
             completed_events,
         }
     }
+
+    pub fn add_expectation(
+        &mut self,
+        event: AccumulatingEvent,
+        voter: &PublicId,
+        proofs: &AccumulatingProof,
+        mut all_voters: BTreeSet<PublicId>,
+    ) {
+        let _ = all_voters.remove(voter);
```

Comment:
Why?
This will make it not shared state as well.

Reply:
My mistake, I will remove it.